### ISSUE TITLE
Add @due(17-12-24)

### ISSUE THREAD

**Issue by tyoc213** [fabiospampinato/vscode-todo-plus#23]
It will be nice if possible to add a `due date` in green more than one week, red in date or after, yellow in the following 7 days... (or some like that)

**Comment by fabiospampinato** [comment]
I like the idea but at the moment I don't have enough time for implementing it.

**Comment by metbril** [comment]
You should modify the title of this issue. It suggests the year to be used without century but instead suggests to add colors for dates. Suggestion:

Add support for color with @due() tags

Color suggestions:

- red for due today
- purple for overdue

Any other date should stay in default tag color for readability of the file.

**Comment by fabiospampinato** [comment]
I think the title is fine, we already format dates that way.

**Comment by fabiospampinato** [comment]
There isn't a `@due` tag just yet, but now you can do this:

<img width="355" alt="screen shot 2018-05-25 at 01 19 55" src="https://user-images.githubusercontent.com/1812093/40518211-c2a87d00-5fb9-11e8-967d-a96d77a876d2.png">

**Comment by sahil48** [comment]
Nicely done. Is there a way to change the @est tag into a tag like @scheduled or @due in the settings?

**Comment by fabiospampinato** [comment]
> Nicely done. Is there a way to change the `@est` tag into a tag like `@scheduled` or `@Due` in the settings?

Nope, that's supposed to be used for time estimates only, and I guess no-one would want to write `@estimate` instead of `@est` 🤔 

> oh god please dont quote me for no reason.

Sorry 😅 It could be worse if your nickname was `@done` though.

**Comment by jods4** [comment]
Nothing prevents us from adding `@due` in tasks, but it would be really cool if it was combined with some visual indicators; a sort command (by due date -- and/or priority); and maybe filtering (although filtering is a generally missing feature, there's another issue about that).

**Comment by fabiospampinato** [comment]
> Nothing prevents us from adding `@due` in tasks, but it would be really cool if it was combined with some visual indicators;

Yes, it's possible to add indicators to the gutter and/or "scrollbar" for this, this issue is basically about adding those since as you said one can already manually write the `@due(whatever)` tag.

**Comment by AdrianWTF** [comment]
about this workaround by @fabiospampinato 
> There isn't a `due` tag just yet, but now you can do this:
> 
> <img alt="screen shot 2018-05-25 at 01 19 55" width="355" src="https://user-images.githubusercontent.com/1812093/40518211-c2a87d00-5fb9-11e8-967d-a96d77a876d2.png">

It has a problem, it sums the remaining days for each task till the desired date
![image](https://user-images.githubusercontent.com/46637889/51081317-e4364200-16ba-11e9-818a-8ffa3f2e4d23.png)
In the example it says 6w for nested to complete. However, from today there are only 3w remaining till feb 3 (the due date for both tasks).
This "due" request would be like a golden feature for me, and to a lot of people too I think. It would be really appreciated. Or even a simple color highlight as sugested by @metbril would be nice. Great work anyway

**Comment by mmickael** [comment]
This feature is needed for tracking tasks.  A similar feature is implemented in Sublime Text package called Plain Tasks.  The feature in Sublime has a popup calendar to choose the date and leave an icon in the gutter of upcoming due tasks.  Will really appreciate having a similar feature in this extension.
